### PR TITLE
Move toolbar functionality from redux to codemirror dispatch

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Editor

### Description
This PR changes the toolbar functionality from redux reducers to code mirror dispatches.
When we used code mirror 5 it was much easier and safer to do this in redux because code mirror 5 was heavily vulnerable for race conditions. Code mirror 6 uses atomic changes. Using this system also eliminates several jump-bugs when using the toolbar.
This is a preparation for the integration of yjs to make it possible that the redux store only replicates the code mirror content.

### Steps

- [x] Added implementation
- [x] Added / updated tests
- [x] I read the [contribution documentation](https://github.com/hedgedoc/react-client/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

**Review by DerMolly — APPROVED** [review]

File: src/components/editor-page/change-content-context/change-content-context.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+/**
+ * Extracts the code mirror reference from the parent context
+ */
+export const useSetCodeMirrorReference = (): SetCodeMirrorReference => {
+  const contextContent = Optional.ofNullable(useContext(changeEditorContentContext)).orElseThrow(
+    () => new Error('No change content received. Did you forget to use the provider component')
+  )
+  return contextContent[1]
+}
```
Comment (ErikMichelson) [inline src/components/editor-page/change-content-context/change-content-context.tsx:44]: Is it required to have these two methods more or less the same except the returned index?

File: src/components/editor-page/editor-pane/hooks/use-handle-upload.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { uploadFile } from '../../../../api/media'
+import { getGlobalState } from '../../../../redux'
+import { supportedMimeTypes } from '../../../common/upload-image-mimetypes'
+import { t } from 'i18next'
```
Comment (ErikMichelson) [inline src/components/editor-page/editor-pane/hooks/use-handle-upload.tsx:10]: Does this use the configured i18next instance?

File: src/components/editor-page/editor-pane/tool-bar/toolbar-button.tsx [hunk 3]
```diff
@@ -8,28 +8,34 @@ import React, { useCallback, useMemo } from 'react'
...
 export interface ToolbarButtonProps {
-  icon: IconName
-  formatType: FormatType
+  i18nKey: string
+  iconName: IconName
+  formatter: GenerateContentEditsCallback
 }
 
-export const ToolbarButton: React.FC<ToolbarButtonProps> = ({ formatType, icon }) => {
+export const ToolbarButton: React.FC<ToolbarButtonProps> = ({ i18nKey, iconName, formatter }) => {
```
Comment (ErikMichelson) [inline src/components/editor-page/editor-pane/tool-bar/toolbar-button.tsx:22]: ESDoc

File: src/components/editor-page/editor-pane/tool-bar/table-picker/table-picker-button.tsx [hunk 4]
```diff
@@ -12,15 +12,32 @@ import { cypressId } from '../../../../../utils/cypress-attribute'
...
+ * @return The created markdown table
+ */
+const createMarkdownTable = (rows: number, columns: number): string => {
+  const rowArray = createNumberRangeArray(rows)
+  const colArray = createNumberRangeArray(columns).map((col) => col + 1)
+  const head = '|  # ' + colArray.join(' |  # ') + ' |'
+  const divider = '| ' + colArray.map(() => '----').join(' | ') + ' |'
+  const body = rowArray.map(() => '| ' + colArray.map(() => 'Text').join(' | ') + ' |').join('\n')
+  return `${head}\n${divider}\n${body}`
+}
```
Comment (ErikMichelson) [inline src/components/editor-page/editor-pane/tool-bar/table-picker/table-picker-button.tsx:39]: Extract this to make the file shorter and the method easier to test

**Review by ErikMichelson — CHANGES_REQUESTED** [review]

File: src/components/editor-page/change-content-context/change-content-context.tsx (hunk 1 above)
Comment (ErikMichelson) [reply src/components/editor-page/change-content-context/change-content-context.tsx:44]: Okay, sounds understandable.

File: src/components/editor-page/editor-pane/hooks/use-handle-upload.tsx (hunk 2 above)
Comment (mrdrogdrog) [reply src/components/editor-page/editor-pane/hooks/use-handle-upload.tsx:10]: Yes! `i18next` is independent from React once it's configured.

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**